### DISCUSSION TITLE
Is it necessary to block HTTPS records for the Plugin Block Unqualified?

### DISCUSSION THREAD

**Discussion by msoxzw** [DNSCrypt/dnscrypt-proxy#2153] [Potential issues]
Currently, the Plugin Block Unqualified blocks A and AAAA records:
https://github.com/DNSCrypt/dnscrypt-proxy/blob/8e46f447999e174dfc448c12c90e2715b399a44c/dnscrypt-proxy/plugin_block_unqualified.go#L34-L36

As more and more browsers and operating systems request HTTPS records, should the Plugin Block Unqualified also block this record type?

**Comment by lifenjoiner** [comment]
That doesn't block `HTTPS` records, try:
```
dig -t https www.cloudflare.com
```

<br>

Unqualified means:
https://github.com/DNSCrypt/dnscrypt-proxy/blob/8e46f447999e174dfc448c12c90e2715b399a44c/dnscrypt-proxy/example-dnscrypt-proxy.toml#L320-L322
For example: "cloudflare" without ".com" which is not a legal internet domain name, and those random strings [Chrome used to query in the past](https://bugs.chromium.org/p/chromium/issues/detail?id=47262).

Append:
`dig -t https cloudflare` will bypass this.
Maybe we can expand the types to be tested, when `HTTPS` record percentage reaches some value.

**Reply by msoxzw** [reply]
Sorry for ambiguity in my issue.

> Append:
> `dig -t https cloudflare` will bypass this.
> Maybe we can expand the types to be tested, when HTTPS record percentage reaches some value.

It is exactly what I would like to say.
